neuromation/cli/parse_utils.py

```diff
@@ -46,3 +53,100 @@ def parse_memory(memory: str) -> int:
 
 def to_megabytes(value: str) -> int:
     return int(parse_memory(value) / (1024 ** 2))
+
+
+@dataclasses.dataclass(frozen=True)
+class JobColumnInfo:
+    id: str
+    title: str
+    align: Align
+    width: ColumnWidth
+
+
+COLUMNS = [
+    JobColumnInfo("id", "ID", Align.LEFT, ColumnWidth()),
+    JobColumnInfo("name", "NAME", Align.LEFT, ColumnWidth(max=20)),
+    JobColumnInfo("status", "STATUS", Align.LEFT, ColumnWidth(max=10)),
+    JobColumnInfo("when", "WHEN", Align.LEFT, ColumnWidth(max=15)),
+    JobColumnInfo("image", "IMAGE", Align.LEFT, ColumnWidth(max=40)),
+    JobColumnInfo("owner", "OWNER", Align.LEFT, ColumnWidth(max=25)),
+    JobColumnInfo("cluster_name", "CLUSTER", Align.LEFT, ColumnWidth(max=15)),
+    JobColumnInfo("description", "DESCRIPTION", Align.LEFT, ColumnWidth(max=50)),
+    JobColumnInfo("command", "COMMAND", Align.LEFT, ColumnWidth(max=100)),
+]
+
+COLUMNS_MAP = {column.id: column for column in COLUMNS}
+
+COLUMNS_RE = re.compile(
+    r"""
+    (?P<col>(\{[^}]+)\})|
+    (?P<sep>\s*(?:,\s*)?)|
+    (?P<miss>.)
+    """,
+    re.VERBOSE,
+)
+COLUMN_RE = re.compile(
+    r"""
+    (?P<id>\w+)
+    (?:
+      (?:;align=(?P<align>\w+))|
+      (?:;min=(?P<min>\w+))|
+      (?:;max=(?P<max>\w+))|
+      (?:;width=(?P<width>\w+))
+    )*
+    (?:;(?P<title>.+))?
+    """,
+    re.VERBOSE,
+)
+
+
+def _get(
+    dct: Dict[str, Optional[str]],
+    name: str,
+    fmt: str,
+    converter: Callable[[str], _T],
+    default: Optional[_T],
+) -> Optional[_T]:
+    val = dct[name]
+    if val is None:
+        return default
+    else:
+        try:
+            return converter(val)
+        except ValueError:
+            raise ValueError(f"Invalid property {name}: {val!r} of format {fmt!r}")
+
+
+def parse_columns(fmt: Optional[str]) -> List[JobColumnInfo]:
+    # Column format is "{id[;field=val][;title]}",
+    # columns are separated by commas or spaces
+    # spaces in title are forbidden
+    if not fmt:
+        return COLUMNS
+    ret = []
+    for m1 in COLUMNS_RE.finditer(fmt):
+        if m1.lastgroup == "sep":
+            continue
+        elif m1.lastgroup == "miss":
+            raise ValueError(f"Invalid format {fmt!r}")
+        column = m1.group()[1:-1]
+        m2 = COLUMN_RE.fullmatch(column)
+        if m2 is None:
+            raise ValueError(f"Invalid format {fmt!r}")
+        groups = m2.groupdict()
+        id = groups["id"]
+        if id not in COLUMNS_MAP:
```

Comment:
I would make column names case-insensitive and allow abbreviations:
```python
id = id.lower()
default = COLUMNS_MAP.get(id)
if default is None:
    for name in COLUMNS_MAP:
        if name.startswith(id):
            if default is not None:
                raise ValueError(f"Ambiguous column {id!r} in format {fmt!r}")
            default = COLUMNS_MAP[id]
    if default is None:
        raise ValueError(f"Unknown column {id!r} in format {fmt!r}")
```
But this can be added in a separate PR.

Reply:
Excellent idea!
Implemented and tested.